openmc/tallies.py

```diff
@@ -1848,6 +1907,116 @@ def swap_filters(self, filter1, filter2, inplace=False):
         if not inplace:
             return swap_tally
 
+    def swap_nuclides(self, nuclide1, nuclide2):
+        """Reverse the ordering of two nuclides in this tally
+
+        This is a helper method for tally arithmetic which helps align the data
+        in two tallies with shared nuclides. This method reverses the order of
+        the two nuclides in place.
+
+        Parameters
+        ----------
+        nuclide1 : Nuclide
+            The nuclide to swap with nuclide2
+
+        nuclide2 : Nuclide
+            The nuclide to swap with nuclide1
+
+        Raises
+        ------
+        ValueError
+            If this is a derived tally or this method is called before the tally
+            is populated with data by the StatePoint.read_results() method.
+
+        """
+
+        # Check that results have been read
+        if not self.derived and self.sum is None:
+            msg = 'Unable to use tally arithmetic with Tally ID="{0}" ' \
+                  'since it does not contain any results.'.format(self.id)
+            raise ValueError(msg)
+
+        # Swap the nuclides in the Tally
+        nuclide1_index = self.nuclides.index(nuclide1)
+        nuclide2_index = self.nuclides.index(nuclide2)
+        self.nuclides[nuclide1_index] = nuclide2
+        self.nuclides[nuclide2_index] = nuclide1
+
+        # Copy the tally data
+        self_mean = copy.deepcopy(self.mean)
```

Comment:
Do you need to deepcopy the entire arrays, or just the portions corresponding to nuclide 1 and 2?